sbc_email/models/import_letters_history.py

```diff
@@ -0,0 +1,438 @@
+                            zip_file = BytesIO(base64.b64decode(
+                                attachment.with_context(bin_size=False).datas))
+                            # catch ALL the exceptions that can be raised
+                            # by class zipfile
+                            try:
+                                zip_ = zipfile.ZipFile(zip_file, 'r')
+                                list_file = zip_.namelist()
+                                # loop over all files in zip
+                                for tmp_file in list_file:
+                                    tmp += (func.check_file(tmp_file) == 1)
+                            except zipfile.BadZipfile:
+                                raise exceptions.Warning(
+                                    _('Zip file corrupted (' +
+                                      attachment.name + ')'))
+                            except zipfile.LargeZipFile:
+                                raise exceptions.Warning(
+                                    _('Zip64 is not supported(' +
+                                      attachment.name + ')'))
+                else:
+                    # files are not selected by user so we find them on NAS
+                    # folder 'Imports'
+
+                    # Retrieve configuration
+                    smb_user = config.get('smb_user')
+                    smb_pass = config.get('smb_pwd')
+                    smb_ip = config.get('smb_ip')
+                    smb_port = int(config.get('smb_port', 0))
+                    if not (smb_user and smb_pass and smb_ip and smb_port):
+                        return False
+
+                    smb_conn = SMBConnection(
+                        smb_user,
+                        smb_pass,
+                        'openerp',
+                        'nas')
+                    config_obj = self.env['ir.config_parameter']
+                    share_nas = (config_obj.search(
+                        [('key', '=', 'sbc_compassion.share_on_nas')])
+                        [0]).value
+                    imported_letter_path = (config_obj.search(
+                        [('key', '=', 'sbc_compassion.scan_letter_imported'
+                          )])[0]).value
+                    if smb_conn.connect(smb_ip, smb_port):
+                        listPaths = smb_conn.listPath(
+                            share_nas,
+                            imported_letter_path)
+                        for sharedFile in listPaths:
+                            if func.check_file(sharedFile.filename) == 1:
+                                tmp += 1
+                            elif func.isZIP(sharedFile.filename):
+                                logger.info(
+                                    'File to retrieve: {}'.format(
+                                        imported_letter_path +
+                                        sharedFile.filename))
+
+                                with tempfile.NamedTemporaryFile(
```

Comment:
You don't need a temporary file, you can directly load the smb file inside a buffer